### PR TITLE
Upgrade to Jekyll 4.x

### PR DESCRIPTION
**Pull request by andrewpmiller**
## What this PR does:

- Upgrades this box of chaos to Jekyll 4.x 🚀
- Presumably improves build performance 🤞
- Hopefully doesn't _break_ everything ~right before a Take 5 blitz~ 🤡

**Preview Build:** https://deploy-preview-433--thegymcms.netlify.app/

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ title: Gymnasium Static Content
 url: https://staging.thegymcms.com
 baseurl:
 timezone: America/New_York
+environment: staging
```
Comment (mbifulco) [inline _config.yml:6]: this is new - it looks like it should be coming from an environment variable, maybe? What does this file do in production environments?

Comment (andrewpmiller) [reply _config.yml:6]: it was missing before...
Netlify uses the different configs for each of the named branch deploys. This config is for staging... it's the default that we use for local builds as well.

production has its own config: https://github.com/gymnasium/gymcms/blob/a6cf1588a212fb533665456568253f81b0f23870/_production_config.yml#L6
as does dev:
https://github.com/gymnasium/gymcms/blob/a6cf1588a212fb533665456568253f81b0f23870/_dev_config.yml#L6

Comment (mbifulco) [reply _config.yml:6]: Got it! Well everything else looks just fine to me! merge when ready 😬

**Review by mbifulco — DISMISSED** [review]

**Review by jgagne — CHANGES_REQUESTED** [review]
Everything looks good. One issue, my doing, GYM-5036 featured state should have been left set to `true` (I believe). @andrewpmiller let me know if it's cool for me to make an update to fix that.

See: https://github.com/gymnasium/gymcms/pull/433/commits/472d4bc52cc3fa53cd1ab7fb35f5f7042b750a99

**Review by jgagne — APPROVED** [review]

**Pull request merged**